akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1522,14 +1522,50 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
    * '''Cancels when''' downstream cancels
    */
   def interleave[T >: Out](that: Graph[SourceShape[T], _], segmentSize: Int): javadsl.Flow[In, T, Mat] =
-    new Flow(delegate.interleave(that, segmentSize))
+    interleave(that, segmentSize, eagerClose = false)
 
   /**
    * Interleave is a deterministic merge of the given [[Source]] with elements of this [[Flow]].
    * It first emits `segmentSize` number of elements from this flow to downstream, then - same amount for `that` source,
    * then repeat process.
    *
-   * After one of upstreams is compete than all the rest elements will be emitted from the second one
+   * Example:
+   * {{{
+   * Source<Integer, ?> src = Source.from(Arrays.asList(1, 2, 3))
+   * Flow<Integer, Integer, ?> flow = flow.interleave(Source.from(Arrays.asList(4, 5, 6, 7)), 2)
+   * src.via(flow) // 1, 2, 4, 5, 3, 6, 7
+   * }}}
+   *
+   * After one of upstreams is complete then all the rest elements will be emitted from the second one, unless
```

Comment:
I'd make this:
"If `eagerClose` is `false` and one of the upstreams complete the elements from the other upstream will continue passing through the interleave stage. If `eagerClose` is `true` and one of the upstream complete interleave will cancel the other upstream and complete itself"

I'd also skip the source samples, since I think the description is enough and code in scaladoc is hard to maintain as it does not get compiled and tested.